lib/versioning/VersionID.js

```diff
@@ -152,4 +156,94 @@ function decode(str) {
     }
 }
 
-module.exports = { generateVersionId, getInfVid, encode, decode };
+/* tiny version Ids constants:
+ *
+ * Note: base62Integer() cannot encode integers larger than 15 digits
+ * so we assume that TV_TOTAL <= 30 and we cut it in half. Please
+ * revisit if TV_TOTAL is greater than 30.
+ */
+const TV_TOTAL = LENGTH_TS + LENGTH_SEQ;
+const TV_HALF = TV_TOTAL / 2;
+const TV_EPAD = '0'.repeat(Math.ceil(TV_HALF * (Math.log(10) / Math.log(62))));
+const TV_DPAD = '0'.repeat(TV_HALF);
+
+/**
+ * Encode a versionId to obscure internal information contained
+ * in a version ID (less than 40 bytes).
+ *
+ * @param {string} str - the versionId to encode
+ * @return {string} - the encoded tinyVersionId
+ */
+function tinyEncode(str) {
+    const part1 = Number(str.substring(0, TV_HALF));
+    const part2 = Number(str.substring(TV_HALF, TV_TOTAL));
+    const part3 = Buffer.from(str.substring(TV_TOTAL));
+    const enc1 = base62Integer.encode(part1);
+    const enc2 = base62Integer.encode(part2);
+    const enc3 = base62String.encode(part3);
+    return (TV_EPAD + enc1).slice(-TV_EPAD.length) +
+        (TV_EPAD + enc2).slice(-TV_EPAD.length) +
+        enc3;
+}
+
+/**
+ * Decode a tinyVersionId. May return an error if the input string is
+ * invalid hex string or results in an invalid value.
+ *
+ * @param {string} str - the encoded tinyVersionId to decode
+ * @return {(string|Error)} - the decoded versionId or an error
+ */
+function tinyDecode(str) {
+    try {
+        let start = 0;
+        const enc1 = str.substring(start, start + TV_EPAD.length);
+        const orig1 = base62Integer.decode(enc1);
+        start += TV_EPAD.length;
+        const enc2 = str.substring(start, start + TV_EPAD.length);
+        const orig2 = base62Integer.decode(enc2);
+        start += TV_EPAD.length;
+        const enc3 = str.substring(start);
+        const orig3 = base62String.decode(enc3);
+        return (TV_DPAD + orig1.toString()).slice(-TV_DPAD.length) +
+            (TV_DPAD + orig2.toString()).slice(-TV_DPAD.length) +
+            orig3.toString();
+    } catch (err) {
+        // in case of exceptions caused by base62 libs
+        return err;
+    }
+}
+
+/**
+ * Encode a versionId to obscure internal information contained
+ * in a version ID.
+ *
+ * @param {string} str - the versionId to encode
+ * @param {boolean} [tiny=false] - use the tiny format (optional)
```

Comment:
could potentially switch to string/enum but it may be overkill. I only mention it cause adding a new default parameter is backwards compatible, but switching a parameter from bool to something else is not.

Reply:
Another comment here is that the brackets around tiny=false is ESLints optional format so the (optional) text at the end is redundant.

Reply:
You are right.
I will rename "tiny" to "base62" as well.